source/docs/reference/graphql-module-definition.md

```diff
@@ -0,0 +1,181 @@
+---
+id: graphql-module-definition
+title: GraphQL module definition
+---
+
+**Front-Commerce’s GraphQL modules** is the mechanism allowing to extend and override any part of the schema defined by other modules.
+
+<blockquote class="info">
+To get started with GraphQL modules, we recommend you to read the [Extend the GraphQL schema](/docs/essentials/extend-the-graphql-schema.html) documentation page.
+</blockquote>
+
+A Front-Commerce GraphQL module has to export an object containing its definition.
+This page contains the API different keys available in a GraphQL module’s definition.
+
+## `namespace`
+
+The simplest (but useless) GraphQL module definition only requires an unique `namespace` key:
+
+```js
+// A minimal Front-Commerce GraphQL module that basically does nothing
+module.exports = {
+  namespace: "Acme/HelloWorld"
+};
+```
+
+This `namespace` can be used by other modules as a dependency, to ensure another module has been registered in the application.
+
+## `modules` (optional)
+
+The `modules` key is useful when creating a
+[meta module](/docs/advanced/graphql/meta-modules.html).
+It allows to declare the submodules to be registered when this module is registered.
+
+It should be an array with a list of submodules to be included in the application.
+
+Example:
+```js
+import Core from "./core";
+import FeatureA from "./feature-a";
+import FeatureB from "./feature-b";
+
+module.exports = {
+  namespace: "Acme/All",
+  modules: [
+    Core,
+    FeatureA,
+    FeatureB
+  ]
+};
+```
+
+<blockquote class="info">
+**Dependency management:** submodules dependencies will be resolved independently.
+
+It means that if a meta module `A` declares `A1` and `A2` in its `modules` key,
+and `A1` depends on module `B` then there is no guarantee that `A2` will be
+initialized _after_ `B`.
+
+You should make dependencies explicit either in the meta module definition or in
+each submodule definition.
```

Comment:
I have noticed there is no new lines created in the actual blockquote.

I'd add also an explanation about our reasoning here. Something like: Our goal is to make sure that each module can be used separatly. Thus, each dependency should be added to the module needing it and not in a meta module.